### ISSUE TITLE
JSON Parsers Error

### ISSUE THREAD

**Issue by ericsaupe** [prismicio/ruby-kit#57]
Hi,

I'm getting an `undefined method 'call' for nil:NilClass` when trying a simple query. The strange thing is that everything works fine using `v1.1.0` but we need to use `>1.3` because we want to take advantage of slices. 

### Code
```
    api = Prismic.api('https://MYDOMAIN.prismic.io/api')
    response = api
        .form('everything')
        .query(Prismic::Predicates::at('document.type', 'page'))
        .submit(api.master_ref)
```

### Relevant Stack Trace
```
/home.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/prismic.io-1.3.2/lib/prismic/json_parsers.rb:197:in `block (2 levels) in group_parser'
/home.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/prismic.io-1.3.2/lib/prismic/json_parsers.rb:197:in `each'
/home.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/prismic.io-1.3.2/lib/prismic/json_parsers.rb:197:in `map'
/home.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/prismic.io-1.3.2/lib/prismic/json_parsers.rb:197:in `block in group_parser'
/home.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/prismic.io-1.3.2/lib/prismic/json_parsers.rb:196:in `each'
/home.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/prismic.io-1.3.2/lib/prismic/json_parsers.rb:196:in `group_parser'
/home.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/prismic.io-1.3.2/lib/prismic/json_parsers.rb:215:in `call'
/home.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/prismic.io-1.3.2/lib/prismic/json_parsers.rb:215:in `fragment_parser'
/home.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/prismic.io-1.3.2/lib/prismic/json_parsers.rb:206:in `block in slices_parser'
/home.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/prismic.io-1.3.2/lib/prismic/json_parsers.rb:205:in `each'
/home.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/prismic.io-1.3.2/lib/prismic/json_parsers.rb:205:in `slices_parser'
/home.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/prismic.io-1.3.2/lib/prismic/json_parsers.rb:215:in `call'
/home.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/prismic.io-1.3.2/lib/prismic/json_parsers.rb:215:in `fragment_parser'
/home.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/prismic.io-1.3.2/lib/prismic/json_parsers.rb:233:in `block in document_parser'
/home.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/prismic.io-1.3.2/lib/prismic/json_parsers.rb:232:in `each'
/home.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/prismic.io-1.3.2/lib/prismic/json_parsers.rb:232:in `map'
/home.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/prismic.io-1.3.2/lib/prismic/json_parsers.rb:232:in `document_parser'
/home.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/prismic.io-1.3.2/lib/prismic/json_parsers.rb:248:in `block in results_parser'
/home.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/prismic.io-1.3.2/lib/prismic/json_parsers.rb:247:in `map'
/home.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/prismic.io-1.3.2/lib/prismic/json_parsers.rb:247:in `results_parser'
/home.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/prismic.io-1.3.2/lib/prismic/json_parsers.rb:262:in `response_parser'
/home.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/prismic.io-1.3.2/lib/prismic.rb:271:in `submit'
```

**Comment by erwan** [comment]
Hi Eric,

Could you tell me what your repository is so I can try it? You can send it through a support ticket if you don't want to post it here.

**Comment by smd686s** [comment]
I think this may be do to the new Separator type.  I hadn't had a chance to submit a PR but here was my attempt at implementing:

https://github.com/SpringfieldCVB/ruby-kit/commit/f9d11481fa8b632fcb6ecbce6c3858aca96aec83

**Comment by n00b2pr0** [comment]
We were attempting to use the sample document mask of `page` generated when creating a new Website Starter repository. Checking against multiple repositories and masks, we narrowed this error down to documents with slices.

We saw the tests were passing, so I compared the JSON from the response_mock against the mask. I noticed the slice zone was in a `blocks` in the response_mock instead of a `body` which was in our repository mask. https://github.com/prismicio/ruby-kit/blob/270858b8ca84e809a3459f954091ddb10355f4fe/spec/responses_mocks/slices.json#L55

I created new mask based off the `page` mask, changed `body` to `blocks`, and was able to return documents.

Hope this helps?
